sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -925,6 +927,22 @@ private void validate() {
       }
     }
 
+    /**
+     * If you want to rename the key of a rule or change its repository or both, register the rule previous repository
+     * and key with this method in order to avoid having issues created with the rule's previous rule key to be closed
+     * and reopened at next analysis.
+     * <p>
+     * Ordering of deprecated rule keys is the same as the calls to this method but is not meaningful to the rule
```

Comment:
@sns-seb could you explain what does this sentence mean?

Reply:
I mean:
Values in `Set` returned by `Rule#deprecatedKeys` are in the same order as they are added to `NewRule` using `addDeprecatedKey`.

Also, I didn't write it, In case a value is added multiple times, it will appear once, in the order of its last addition to `NewRule`

Reply:
also, there is some constraints on the deprecated keys themselves, which I forgot to document. I'll add it.